site-permissions/site-permissions-impl/src/main/java/com/duckduckgo/site/permissions/impl/SitePermissionsDialogActivityLauncher.kt

```diff
@@ -446,10 +448,19 @@ class SitePermissionsDialogActivityLauncher @Inject constructor(
     }
 
     private fun onResultMultipleSystemPermissionsRequest(grantedPermissions: Map<String, Boolean>) {
-        if (grantedPermissions.values.contains(false)) {
-            systemPermissionDenied()
+        val locationPermissions = setOf(Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION)
+
+        val permissionsGranted = if (grantedPermissions.keys == locationPermissions) {
+            // For location permissions request, it is enough for the user to grant access to approximate location
+            grantedPermissions.getValue(Manifest.permission.ACCESS_COARSE_LOCATION)
         } else {
+            !grantedPermissions.containsValue(false)
+        }
```

Comment:
NIT: Since this logic is specific to system permissions and we already have access to `systemPermissionsHelper`, it would be cleaner to move this logic to its interface. This helps keep this part of the code simpler and more focused.